### PR TITLE
74954 custom supporting documentation component for form 10-10d

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

This PR introduces a custom page supporting document component for use within the 10-10d form. It provide an interactive file upload needed for early-stage user testing and it allows extra customization of the file upload interface which will be a key piece in 10-10d. This is a first pass; the component will be heavily modified in the future.

- Added custom supporting docs upload page component
- Added unit test coverage for new component
- **Team**: IVC-CHAMPVA
- **Flipper**: Not in use, form not set to go to prod

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/74954

## Testing done

- Manual testing to verify the component behaves as expected
- Introduced new unit tests to serve as a baseline
- Ran `yarn test:coverage-app ivc-champva/10-10D` to verify new and existing tests run and pass

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        | Screenshot 2024-02-07 at 12 24 22 |
| Desktop |        | Screenshot 2024-02-07 at 12 24 14 |

## What areas of the site does it impact?

This form only.

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

NA

### CODE REVIEW

**Review by cloudmagic80 — DISMISSED** [review]
I approve!

File: src/applications/ivc-champva/10-10D/helpers/utilities.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const getFileSize = num => {
+  if (num > 999999) {
+    return `${(num / 1000000).toFixed(1)} MB`;
+  }
+  if (num > 999) {
+    return `${Math.floor(num / 1000)} KB`;
+  }
+  return `${num} B`;
+};
```
Comment (balexandr) [inline src/applications/ivc-champva/10-10D/helpers/utilities.js:9]: Is there any reason we aren't using `else if` and do we not need a `toFixed()` for the KB?
```export const getFileSize = num => {
  if (num > 999999) {
    return `${(num / 1000000).toFixed(1)} MB`;
  } else if (num > 999) {
    return `${(num / 1000).toFixed(1)} KB`;
  } else {
    return `${num} B`;
  }
};

File: src/applications/ivc-champva/10-10D/tests/config/form.unit.spec.jsx [hunk 2]
```diff
@@ -44,6 +59,59 @@ testNumberOfWebComponentFields(
...
+
+    await waitFor(() => {
+      expect($('.no-attachments', view.container)).to.exist;
+    });
+  });
+});
+
+describe('File sizes', () => {
+  it('should be in bytes for values < 999', () => {
+    expect(getFileSize(998) === '998 B').to.be.true;
```
Comment (balexandr) [inline src/applications/ivc-champva/10-10D/tests/config/form.unit.spec.jsx:105]: Using `to.equal()` might increase readability.
`expect(getFileSize(998)).to.equal('998 B');`

**Review by balexandr — DISMISSED** [review]

File: src/applications/ivc-champva/10-10D/helpers/utilities.js (hunk 1 above)
Comment (michaelclement) [reply src/applications/ivc-champva/10-10D/helpers/utilities.js:9]: I pulled it from elsewhere in the codebase, so not sure why it is the way that it is, but your suggestion is definitely a lot more uniform.

**Comment by IGallupSoCo** [comment]
This looks good to me, I'm just withholding approval until main gets pulled in to fix the CI failures.

**Comment by michaelclement** [comment]
> From the FE CoP side, this looks good, I'm just withholding approval until main gets pulled in to fix the CI failures.

Pulled from main and CI failures have been resolved 👍

**Review by micahchiang — APPROVED** [review]

**Pull request merged**